sql/moz-fx-data-shared-prod/org_mozilla_ios_firefox_derived/legacy_metrics_v1/query.sql

```diff
@@ -0,0 +1,206 @@
+      labeled_counter(
+        reading_list_add,
+        ["reader-mode-toolbar", "share-extension", "page-action-menu"]
+      )
+    ) AS labeled_counter_reading_list_add,
+    ANY_VALUE(
+      labeled_counter(reading_list_delete, ["reader-mode-toolbar", "reading-list-panel"])
+    ) AS labeled_counter_reading_list_delete,
+    SUM(
+      IF(object = "reading-list-item" AND method = "open", value, 0)
+    ) AS counter_reading_list_open,
+    SUM(
+      IF(object = "reading-list-item" AND method = "mark-as-read", value, 0)
+    ) AS counter_reading_list_mark_read,
+    SUM(
+      IF(object = "reading-list-item" AND method = "mark-as-unread", value, 0)
+    ) AS counter_reading_list_mark_unread,
+    SUM(IF(object LIKE "qr-code%" AND method = "scan", value, 0)) AS counter_qr_code_scanned,
+  FROM
+    extracted_event
+  JOIN
+    labeled
+  USING
+    (client_id, submission_date)
+  GROUP BY
+    client_id,
+    submission_date
+)
+SELECT
+  submission_timestamp,
+  (SELECT AS STRUCT metadata.* EXCEPT (uri)) AS metadata,
+  normalized_app_name,
+  normalized_channel,
+  normalized_country_code,
+  normalized_os,
+  normalized_os_version,
+  STRUCT(
+    client_id,
+    CAST(NULL AS string) AS android_sdk_version,
+    metadata.uri.app_build_id AS app_build,
+    metadata.uri.app_update_channel AS app_channel,
+    metadata.uri.app_version AS app_display_version,
+    arch AS architecture,
+    "Apple" AS device_manufacturer,
+    device AS device_model,
+    locale,
+    os,
+    osversion AS os_version,
+    CAST(NULL AS string) AS telemetry_sdk_build
+  ) AS client_info,
+  STRUCT(
+    STRUCT(
+      string_search_default_engine AS search_default_engine,
+      string_preferences_mail_client AS preferences_mail_client,
+      string_preferences_new_tab_experience AS preferences_new_tab_experience
+    ) AS string,
+    STRUCT(
+      counter_tabs_cumulative_count AS tabs_cumulative_count,
+      counter_glean_validation_foreground_count AS glean_validation_foreground_count,
+      counter_reader_mode_open AS reader_mode_open,
+      counter_reader_mode_close AS reader_mode_close,
+      counter_reading_list_open AS reading_list_open,
+      counter_reading_list_mark_read AS reading_list_mark_read,
+      counter_reading_list_mark_unread AS reading_list_mark_unread,
+      counter_qr_code_scanned AS qr_code_scanned
+    ) AS counter,
+    STRUCT(
+      labeled_counter_search_counts AS search_counts,
+      labeled_counter_bookmarks_view_list AS bookmarks_view_list,
+      labeled_counter_bookmarks_open AS bookmarks_open,
+      labeled_counter_bookmarks_add AS bookmarks_add,
+      labeled_counter_bookmarks_delete AS bookmarks_delete,
+      labeled_counter_tabs_open AS tabs_open,
+      labeled_counter_tabs_close AS tabs_close,
+      labeled_counter_reading_list_add AS reading_list_add,
+      labeled_counter_reading_list_delete AS reading_list_delete
+    ) AS labeled_counter
+  ) AS metrics
+FROM
+  aggregated
+FULL JOIN
```

Comment:
Aggregated was already created from extract_event, why do we need to full join here?

Reply:
Good point, I added this here because of the full join that is necessary on the derived core table. This full join is not necessary.

Reply:
IIUC, this will only run once, so we do not need to spend time optimizing if it's correct.